src/main/java/com/teamacronymcoders/epos/pathfeature/item/ItemRewardFeature.java

```diff
@@ -0,0 +1,44 @@
+package com.teamacronymcoders.epos.pathfeature.item;
+
+import com.teamacronymcoders.epos.api.characterstats.ICharacterStats;
+import com.teamacronymcoders.epos.pathfeature.PathFeature;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.item.ItemEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+
+public class ItemRewardFeature extends PathFeature {
+    private final String identifier;
+    private final ItemStack stack;
+
+    public ItemRewardFeature(String identifier, ItemStack stack) {
+        super(new TranslationTextComponent("pathfeature.epos.itemreward.name", stack.getDisplayName()),
+                new TranslationTextComponent("pathfeature.epos.itemreward.description", stack.getDisplayName()));
+        this.identifier = identifier;
+        this.stack = stack;
+    }
+
+    @Override
+    public void applyTo(LivingEntity character, ICharacterStats characterStats) {
+        World world = character.world;
+        CompoundNBT entityNBT = character.getEntityData();
+        if (entityNBT.contains("item_rewards") && !entityNBT.getCompound("item_rewards").contains(identifier)) {
+            entityNBT.getCompound("item_rewards").putBoolean(identifier, true);
+            ItemEntity entity = new ItemEntity(world, character.posX, character.posY, character.posZ);
+            entity.setItem(stack);
+            character.world.addEntity(entity);
+        } else {
+            CompoundNBT newNBT = new CompoundNBT();
+            newNBT.putBoolean(identifier, true);
+            ItemEntity entity = new ItemEntity(world, character.posX, character.posY, character.posZ);
```

Comment:
I think these three lines `ItemEntity entity =` through `character.world.addEntity` can just be moved after the if/else. Though I am not quite sure what `newNBT` is supposed to be used for given currently it doesn't seem to be doing anything.

Reply:
Why are you spawning an entity and not just trying to give it to the player.